include/PROPOSAL/particle/Particle.h

```diff
@@ -119,20 +119,20 @@ struct Loss {
 };
 
 struct StochasticLoss : public Loss {
-    StochasticLoss(int, double, Vector3D, Vector3D, double, double, double);
-    Vector3D position;
-    Vector3D direction;
+    StochasticLoss(int, double, const Vector3D&, const Vector3D&, double, double, double);
+    Cartesian3D position;
+    Cartesian3D direction;
     double time;
     double propagated_distance;
     double parent_particle_energy;
 };
 
 struct ContinuousLoss : public Loss {
-    ContinuousLoss(std::pair<double, double>, std::pair<Vector3D, Vector3D>,
-            std::pair<Vector3D, Vector3D>, std::pair<double, double>);
+    ContinuousLoss(std::pair<double, double>, std::pair<const Vector3D&, const Vector3D&>,
```

Comment:
I am not quite sure if the parameter should not be a 
`std::pair<Vector3D, Vector3D> const&`

Reply:
A pair must not hold references. If you really need a pair of references, you need to use
https://en.cppreference.com/w/cpp/utility/functional/reference_wrapper